### DISCUSSION TITLE
Prerendering with target=static and ssr=true

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#8837] [Questions (Nuxt 2)] [answered]
I like the feature of prerendering all my pages/routes in combination with `generate.routes()` inside `nuxt.config.js`. This way I can prerender a whole website and it’s prepared for search engine crawlers … 

I’ve set up a little test app where content comes from an API-CMS. It works fine, but as soon as I am changing content through the API-CMS, it won’t appear in my nuxtjs-website, after executing `npm run generate`. I had a look at the `dist` folder and found out that the new data is not used inside the prerendered html. If I wait for 20–60 seconds, then `npm run generate` will work, if I trigger it manually.

But that’s a problem, because after every change the API-CMS triggers a "webhook", that will execute `npm run generate`. But then the new data is not in the prerendered files …

Is this a problem with nuxtJS — maybe there is a cache mechanism built in?
Or is it a problem with the hosting or with the CMS?

How would you deal with this?

**Comment by danielroe** [comment, marked as answer]
I would expect that a webhook calling Nuxt to rerun `nuxt generate` would be successful in updating the pages with new content. Possible issues to investigate include'
1. whether your CMS is returning the new data to API calls (perhaps it hasn't yet propagated by the point that Nuxt generate is running)
2. whether you are indeed fetching the data with asyncdata/fetch
3. whether you have a cache on your web host that is delaying the update

**Reply by ghost** [reply]
@danielroe Thx for your suggestions. I’ve found out that the API-CMS caches requests. Fortunately there is an option to disable caching. So my intention now is to disable caching only while re-generating the nuxtJS-app. This leads me to the question:

How can I determine if Nuxt is `currently generating` and not viewed in the browser, so that I can adjust the cache-option of my API-CMS?

**Reply by danielroe** [reply]
@JonathanDoelan What you want to do is best done from a couple of Nuxt hooks. See [generate hooks documentation](https://nuxtjs.org/docs/2.x/internals-glossary/internals-generator#hooks). You could, for example, disable caching in `generate:before` and re-enable it on `generate:done`.
